### ISSUE TITLE
Firefox support

### ISSUE THREAD

**Issue by mokkin** [jitsi/jitsi-meet#4]
This is just a support request.

Chromium works very well.
Please support also other browsers such as Firefox in the future.

Thank you for your great work!

**Comment by palant** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=784517 has been resolved as a duplicate of https://bugzilla.mozilla.org/show_bug.cgi?id=1095218 - and the later has been resolved as fixed last Thursday. Unfortunately, the code in question uses browser detection rather than feature detection, so the current Firefox nightlies will still redirect to `chromeonly.html`.

**Comment by emcho** [comment]
Hey @palant . Support in Meet is still being worked on and the corresponding support in firefox is still being stabilised. We will obviously change the browser check once we actually support another browser

**Comment by Zalmoxisus** [comment]
I see that commit 05bbfda5bb6b54ff78dec93ede3d5054ad49843a now adds support for firefox multistream. Is it for FF 38 Aurora/DevEd?

**Comment by TabeebOsama** [comment]
Was FF support added in release 2.8?  Thank you!

**Comment by bgrozev** [comment]
@TabeebOsama the recent 2.8 release was of Jitsi (the desktop client), not Jitsi Meet. Firefox support was recently added to Jitsi Meet, but AFAIK it is still experimental.

**Comment by gpolitis** [comment]
As @bgrozev FF support is still experimental but it is enabled by default for [Firefox Nightly](https://nightly.mozilla.org/) users.

**Comment by TabeebOsama** [comment]
Thank you for your prompt replies.  I know this is not this thread.  However, I have two more questions related:
1- Which IE plaug-n does Jitsi support and you recommend?
2- Will you support Bowser?

Thanks,

**Comment by emcho** [comment]
@TabeebOsama  please send your question to the mailing list.

**Comment by dev-mraj** [comment]
any workarround?

can't we merge audio+video streams into one stream..? will it work if enableFirefoxSupport=true

**Comment by gpolitis** [comment]
@dev-mraj any workaround for what exactly?

**Comment by dev-mraj** [comment]
Firefox support

**Comment by gpolitis** [comment]
adding "enableFirefoxSupport: true" in config.js should work.

**Comment by TabeebOsama** [comment]
Is this for nightly Firefox or all Firefox?

This email message is for the sole use of the intended recipient(s) and may contain confidential and privileged information. If you are not the intended recipient, please contact the sender by reply email and destroy all copies of the original message

**Comment by gpolitis** [comment]
Firefox support only works for Firefox Nightly (v39), at least for now. We might enable it for Aurora/Dev edition (v38) as well but it's not going to work with stable (v37).

**Comment by foss-** [comment]
+1 for enabling Aurora / Dev v38.

**Comment by h0nIg** [comment]
whats the status of "We might enable it for Aurora/Dev edition (v38)"?

**Comment by gpolitis** [comment]
It is actually enabled for v38 (now Beta). We're still blocked by [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1155246) though.

**Comment by gpolitis** [comment]
FF v40 (now Aurora) includes some important fixes and it seems to be working relatively well, if people want to test.

**Comment by foss-** [comment]
George, what's up with https://bugzilla.mozilla.org/show_bug.cgi?id=1155246 ? Is it still a valid bug? Your last comment seems to indicate otherwise.

**Comment by gpolitis** [comment]
My last comment still applies. I'm not sure if it's gone (or if it ever existed), but it doesn't seem to affect meet anymore.

**Comment by gpolitis** [comment]
Thank for trying this out! I'm not able to reproduce this on https://beta.meet.jit.si/. Could you please paste the JS console log here?

**Comment by foss-** [comment]
Sent to you via encrypted mail.

**Comment by gpolitis** [comment]
As per our private conversation, the problem appears to be that you try to enter the conference without sharing your camera. I've noted this in #271. Also, screen sharing doesn't currently work. That's documented in #296.

**Comment by ShalokShalom** [comment]
Means this, that it will work with 40 (Aurora) 

https://hacks.mozilla.org/2015/06/firefox-multistream-and-renegotiation-for-jitsi-videobridge

**Comment by wolftune** [comment]
This *does* now work with FF 40. This should probably be closed, but FF isn't fully on-par until #296 is done (and maybe some other items)

**Comment by geileszeuch** [comment]
Is there any chance that this will work on Firefox for android? Currently it shows the not-supported screen after typing in a room name.

**Comment by wolftune** [comment]
@geileszeuch are you on FF 40 ? ( I don't know if that works, but I know it *won't* work if on older)

**Comment by geileszeuch** [comment]
I am FF 40.

**Comment by gpolitis** [comment]
@geileszeuch Unfortunately I don't think FF for Android has support for WebRTC, so we can't support it even if we wanted to. OTOH the performance would be absolutely abysmal, unless it runs on powerful hardware; so my thinking is that native is the way to go for mobile devices.

**Comment by geileszeuch** [comment]
I am sorry, I had my user agent edited. It works totally fine on Android and the performance is also quite good on my phone. Thank you guys. Its time to say goodbye to Skype.

**Comment by wolftune** [comment]
@geileszeuch great to know! I should try that myself

**Comment by gpolitis** [comment]
This is unexpectedly good news, thanks @geileszeuch for sharing your experience with us. Could you please share what tablet/smartphone you're running meet on?

**Comment by TabeebOsama** [comment]
This will be an important feature to add.

Thank you,
Osama

**Comment by marclaporte** [comment]
I just tested audio / videoconference at https://meet.jit.si/ using 
1- Galaxy Note 3 with Firefox 40 installed with F-Droid 
2- Firefox 40.0.2 on Windows 8

It all seems to work fine. Congrats!  This is very very exciting!

Jitsi Meet is not responsive though. I brought this up on the dev mailing list:
http://lists.jitsi.org/pipermail/dev/2015-August/025018.html

Thanks!

**Comment by geileszeuch** [comment]
I used an Nexus 5. The user interface is a little tiny, but I am happy it works.

**Comment by marclaporte** [comment]
Hi!

So general Firefox support is resolved AFAICT.

I added a PR to add Firefox to supported browsers page:
https://github.com/jitsi/jitsi-meet/pull/354

I realize some issues are being discovered such as:
https://github.com/jitsi/jitsi-meet/issues/343
But these should be treated as browser-specific bugs (each with their own bug report), just like if they affected another browser.

Thanks!

**Comment by foss-** [comment]
Agree w @marclaporte. Why is this still open?

**Comment by gpolitis** [comment]
Thank you for the reminder.

**Comment by knocte** [comment]
AFAIU if you close this issue, it should mean that I can access http://meet.jitsi.org/ with my Firefox browser. But I can't. The version I'm using is 41.0 (in Ubuntu 14.04.1).

**Comment by gpolitis** [comment]
Can you access https://meet.jit.si/ with FF? I'm not sure what version of meet we're running on https://meet.jit.si/, probably an older one.

**Comment by knocte** [comment]
> Can you access https://meet.jit.si/ with FF?

Yes that seems to work.

> I'm not sure what version of meet we're running on https://meet.jit.si/, probably an older one.

Can you upgrade it or make it to rather redirect to https://meet.jit.si/?

**Comment by gpolitis** [comment]
Good to hear that it works for you and I'm sorry for the confusion! I'll try to find out what the situation is with m.j.o.

**Comment by KuenzelIT** [comment]
Hey,
so Firefox is supported now? Does not work for me with 43.0.4 of Firefox.
What are the supported browsers right now?
Thanks!

**Comment by foss-** [comment]
I agree the ticket situation is confusing. This was closed when #45 was created. Then #45 was closed referring that FF issues are now tracked in #4 (despite #4 already being closed). That logical issues aside, in my tests, Firefox (which you should update to v44 since that has 9 security fixes you should want on your system) works fine using meet.jit.si.

**Comment by gpolitis** [comment]
@KuenzelIT feel free to drop us an email if you're experiencing problems with FF. As @foss- is saying, it should be working fine right now.

**Comment by KuenzelIT** [comment]
Still does not work, this is how it looks for me:
http://s10.postimg.org/ybbv4zhq1/Meet_Jitsi_FF.png

Can't really do anything there. It works in IE 11.

Maybe it is something with my FF specifically, there are some rare sites, where nothing seems to work. Maybe I'm going to try it without add-ons...

**Comment by bgrozev** [comment]
> Still does not work, this is how it looks for me:

Open up the java scropt console and look for errors.

**Comment by KuenzelIT** [comment]
These are the errors in the console:
```
This appears to be Firefox, ver: 44 lib-jitsi-meet.min.js:3:27489

ReferenceError: mozRTCSessionDescription is not defined lib-jitsi-meet.min.js:4:29512

mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create app.bundle.min.js:11:11895

ReferenceError: JitsiMeetJS is not defined app.bundle.min.js:7:24492
```

Does this help? I don't have a clue what's wrong there.

**Comment by gpolitis** [comment]
Thank you @KuenzelIT. Have you by any chance disabled WebRTC in Firefox?

**Comment by palant** [comment]
For reference, the relevant `about:config` preference is called `media.peerconnection.enabled`. It cannot be changed via Firefox settings but [this add-on](https://addons.mozilla.org/addon/happy-bonobo-disable-webrtc/) will change it for example. With `media.peerconnection.enabled` set to `false` the `mozRTCSessionDescription` constructor will indeed be undefined.

**Comment by KuenzelIT** [comment]
Looks like it was disabled. `media.peerconnection.enabled` was set to false, altough i don't remember changing it. I could change it without an add-on

Thanks for your help!

**Comment by gpolitis** [comment]
You're welcome @KuenzelIT. I'm locking this conversation because any bugs in our FF support should be entered as new issues and questions about our FF support should be addressed in our mailing list.
